DataGrid: Fix navigateToRow docissues (#1354)
DevExpress/devextreme-documentation: api-reference/10 UI Widgets/GridBase/3 Methods/navigateToRow(key).md

---
---
##### shortDescription

> Navigates the grid to the data page that contains the row with the specified key, expand groups if necessary, and scrolls the grid to display the row if it is outside the viewport.

Reviewer: "expand groups if necessary" Why 'if  necessary'?

Author: Because groups will be expanded only if data should be grouped and the group with the target row is collapsed. If there's no grouping or the group is already expanded, then the method won't expand it.

Reviewer: But you cannot state 'if necessary' because this is not some necessity. If something happens in certain cases, then you should mention these cases, not that they happen when it is necessary.

The following requirements apply when you use this method: 

- The widget's [keyExpr](/api-reference/10%20UI%20Widgets/dxDataGrid/1%20Configuration/keyExpr.md '{basewidgetpath}/Configuration/#keyExpr') or the [store's](/concepts/70%20Data%20Binding/5%20Data%20Layer/1%20Creating%20DataSource/3%20What%20Are%20Stores.md '/Documentation/Guide/Data_Binding/Data_Layer/#Creating_DataSource/What_Are_Stores') **key** option should be specified.

> - Rows should be initially sorted by keys. They can be sorted on the client using a column's [sortOrder](/api-reference/_hidden/GridBaseColumn/sortOrder.md '{basewidgetpath}/Configuration/columns/#sortOrder') or the **DataSource**'s [sort](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/sort.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#sort') option, or come already sorted from the server.

Reviewer: ```suggestion
- Rows should be sorted by keys initially. You can use a column's [sortOrder](/api-reference/_hidden/GridBaseColumn/sortOrder.md '{basewidgetpath}/Configuration/columns/#sortOrder') or the **DataSource**'s [sort](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/sort.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#sort') option to sort them on the client, or come sorted from the server.
```

Author: ```suggestion
- Rows should be initially sorted by keys. You can sort them on the server or use a column's [sortOrder](/api-reference/_hidden/GridBaseColumn/sortOrder.md '{basewidgetpath}/Configuration/columns/#sortOrder') or the **DataSource**'s [sort](/api-reference/30%20Data%20Layer/DataSource/1%20Configuration/sort.md '/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#sort') option to sort the rows on the client.
```

Reviewer: Ok

> - [Scrolling mode](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/scrolling/#mode) should be different from *"infinite"*.

Reviewer: ```suggestion
- [Scrolling mode](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/scrolling/#mode) should not be *"infinite"*.
```
